zerver/lib/rate_limiter.py

```diff
@@ -623,6 +634,33 @@ def rate_limit_endpoint_absolute(endpoint_name: str) -> None:
         raise RateLimitedError(secs_to_freedom)
 
 
+def get_ipv6_network_portion(ipv6: str, prefix: int) -> str:
+    """Returns the standard hex representation of the NETWORK portion
+    of that ipv6 address, omitting leading zeros and substituting
+    consecutive zero-groups with double colons.
+    This essentially tells us which bucket should this ipv6 belong to.
+
+    Examples:
+
+    For a given ipv6 = 2001:0db8:ce1:12::8a2e:0370
+
+    >>> get_network_prefix(ipv6, 64)
+    '2001:db8:ce1:12::/64'
+
+    >>> get_network_prefix(ipv6, 56)
+    '2001:db8:ce1::/56'
+
+    >>> get_network_prefix(ipv6, 48)
+    '2001:db8:ce1::/48'
+
+    Note if ipv6 has an early ::
+    >>> get_network_prefix('2001:0db8::ce1:12:8a2e:0370', 64)
+    '2001:db8::/64'
+    """
+
+    return str(IPv6Network(ipv6).supernet(new_prefix=prefix))
```

Comment:
This comment block's examples now disagree with the name of the function.  Given that all it's doing now is calling a stdlib function, and there is exactly one callsite, and we don't test this function separately -- I'd just inline the definition and do away with the long comment block.

Reply:
> This comment block's examples now disagree with the name of the function

How is that ? the function name is `get_ipv6_network_portion` and the examples show exactly that, `2001:db8:ce1:12::/64` is indeed the network portion.

> we don't test this function separately

We actually test it, just not directly. The assertions in `test_ip_bucket_key` make sure that the key, which is mainly constructed by `get_ipv6_network_portion`, should be as expected

Reply:
The examples show calling a function named `get_network_prefix`.  The function is currently named `get_ipv6_network_portion`.  Those are ... different. :)

I wasn't saying that you needed to test it separately, or implying that it was untested -- just that _one_ potential reason that we might have a function with only one callsite is to make testing of just that function in isolation easier, or to make it easier to mock out when testing other portions.  Neither of those reasons arguing for a separate function are present here, that's all I meant.